### PR TITLE
Improvement speed search package manager

### PR DESCRIPTION
**Pull request by chuongmep**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Just small update for user to look search is fast.

IPltaiVwtB

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

@QilongTang 

### FYIs

@Amoursol

### CODE REVIEW

File: src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs [hunk 1]
```diff
@@ -5,21 +5,37 @@
...
     {
+        private DispatcherTimer delayTimer;
         /// <summary>
         /// Constructor
         /// </summary>
         public SearchBoxControl()
         {
             InitializeComponent();
         }
+        public static DispatcherTimer Debounce(DispatcherTimer dispatcher, TimeSpan interval, Action action)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs:26]: `internal` please

File: src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs [hunk 2]
```diff
@@ -29,14 +45,15 @@ public SearchBoxControl()
...
         private void SearchTextBox_OnTextChanged(object sender, TextChangedEventArgs e)
         {
-            var textBox = sender as TextBox;
-            if (textBox == null) return;
-
-            (this.DataContext as PackageManagerSearchViewModel).SearchAndUpdateResults(textBox.Text);
-
-            textBox.Focus();
+            // set delay for this event 500ms
+            Debounce(delayTimer, TimeSpan.FromMilliseconds(500), () =>
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs:49]: Would you move the `500` out as an internal static int property?

Comment (chuongmep) [reply src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs:49]: I guess I will more this more to private, it is good to go ?

File: src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs (hunk 1 above)
Comment (chuongmep) [reply src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs:26]: I guess I will more this method to private, it is good to go ?

File: src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs (hunk 2 above)
Comment (chuongmep) [reply src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs]: Sure, this one resolved.

File: src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs (hunk 1 above)
Comment (chuongmep) [reply src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs]: I moved to private.

**Comment by QilongTang** [comment]
@zeusongit It's weird that this simple PR would report that 3 new files added.

**Review by QilongTang — APPROVED** [review]

**Comment by zeusongit** [comment]
> @zeusongit It's weird that this simple PR would report that 3 new files added.

.\en-US\fallback_docs\CoreNodeModels.Formula_img.jpg    Added
.\en-US\fallback_docs\CoreNodeModels.Formula.dyn        Added
.\en-US\fallback_docs\CoreNodeModels.Formula.md         Added

It is pointing at these 3 files.

**Pull request merged**